blockchain/src/history/mmr_store.rs

```diff
@@ -190,6 +191,14 @@ impl<'a, 'env> LightMMRStore<'a, 'env> {
         block_number: u32,
     ) -> Self {
         let size = get_size(hist_tree_table, tx, block_number);
+
+        // If size is 0 we need to copy the entries from the previous block number to the new one
+        if size == 0 {
+            init_mmr_from_block_number(hist_tree_table, tx, block_number);
+        }
```

Comment:
For the light history store, what we are using are "peaks-only MMRs" 
However, this means that if we have to revert blocks, we don't have enough information to just remove txns and go back to a previous state, as we do with the regular MMR.

So what we are doing is to store "peaks only mmrs" per block, i.e. essentially we have "snapshots" of how the peaks only MMR looked at each block. 

However, since history nodes construct a big MMR with all the transactions of the given epoch, and this is the root that we need to verify against, then it means the starting point for block N is the peaks only mmr of block N-1, which is what this line is doing:
when we start pushing transactions for a block that we didn't have before, we first need to start with the peaks only mmr of the previous block and continue from there.

Let me know if this clarifies just question